src/lib/homestore_backend/index_kv.cpp

```diff
@@ -59,9 +59,98 @@ BlobManager::NullResult HSHomeObject::add_to_index_table(shared< BlobIndexTable
     return folly::Unit();
 }
 
+BlobManager::Result< homestore::MultiBlkId > HSHomeObject::move_to_tombstone(shared< BlobIndexTable > index_table,
+                                                                             const BlobInfo& blob_info) {
+    // since there may be a crash at any step of this function, we need to make sure the index table is always
+    // correct when we recover and reapply the raft log.
+    // the sequence here must be get, put and remove, so that we can keep the correctness of the index table.
+
+    BlobRouteKey index_key{BlobRoute{blob_info.shard_id, BlobRoute::KeyType::kTypeNormal, blob_info.blob_id}};
+    BlobRouteValue index_value;
+    homestore::BtreeSingleGetRequest get_req{&index_key, &index_value};
+    auto status = index_table->get(get_req);
+    if (status != homestore::btree_status_t::success) {
+        LOGERROR("Failed to get from index table {}", index_key.to_string());
+        return folly::makeUnexpected(BlobError::UNKNOWN_BLOB);
+    }
+
+    index_key.change_to_tombstone();
+    // we set a tombstone in key, of which GC can take the benifit to quickly finding the delete blkIDs by using
```

Comment:
I fully get the point of this approach, compared to update the value not the key, it provide a fast way to get active/deleted blobs set of a shard.

However I am not convinced by the tradeoff,  with this approach we takes 4(include journal)+( + in the case that append/delete requires b-tree splits) index update compared to 2 if we only change the value, for every blob delete.  
The gain is saving the (maybe one time, only when GC) cost of filtering out deleted blob(identify from index value) in memory, which looks to me like cheaper

Reply:
I do not think it is acceptable that we have to scan the whole table and compare the value one by one when GC any shard, especially when the index table is very large . 

i get your point. only updating the value will not change the btree structure and will schedule less index update operations. 

we can discuss this tradeoff offline with more detail.